### PR TITLE
Guest view

### PR DESCRIPTION
**Pull request by luukaass**

### CODE REVIEW

File: app/Tools/Query.php [hunk 1]
```diff
@@ -195,4 +212,74 @@ public static function getMessageCount()
         return count($notifications);
     }
 
-}
\ No newline at end of file
+
+    public static function getSessionEventsMonth($oDay)
```
Comment (tarjmp) [inline app/Tools/Query.php]: This method is equal to getUserEventsMonth, except for the query part -> duplicate code

File: resources/lang/en/calendar.php [hunk 2]
```diff
@@ -22,5 +22,7 @@
     'saturday'  => 'Saturday',
     'sunday'    => 'Sunday',
 
-    'show-rejected' => 'Display rejected events'
+    'show-rejected' => 'Display rejected events',
+
+    'no_groups' => 'No public groups found',
```
Comment (tarjmp) [inline resources/lang/en/calendar.php]: Same text is also present in group language file, so no need to add it here

File: resources/views/public-group.blade.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                                                    <label class="custom-control-label float-left"
+                                                           for="same-address-{{$group->id}}">{{$group->name}}</label>
+                                                </div>
+                                            </h3>
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+                        @empty
+                            {{ __('calendar.no_groups') }}
```
Comment (tarjmp) [inline resources/views/public-group.blade.php]: Could be group.no_groups

File: resources/views/layouts/app.blade.php [hunk 4]
```diff
@@ -72,6 +72,15 @@
 
                 <!-- Right Side Of Navbar -->
                 <ul class="navbar-nav ml-auto">
+                    <form method="GET" action="{{ route('search') }}" class="mr-5">
```
[the remark this answers is not in the document]
Comment (luukaass) [reply resources/views/layouts/app.blade.php]: The search permission check only checks if a user is logged in. This permission is removed by 0326c77a4771302ab10e2ad0e15060e80d45aaf5, as the search functionality only returns search results the guest/user can see (methods getAllAccessibleEvents and getAllAccessibleGroups).

**Pull request merged**